### ISSUE TITLE
Collection::offsetGet stub too strict: expects int<0, max> instead of int

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#499]
## Description

The `Illuminate\Support\Collection::offsetGet` stub parameter type is `int<0, max>` but Laravel's actual signature accepts plain `int` (or `mixed` via `ArrayAccess`). This causes false-positive `ArgumentTypeCoercion` when accessing collection elements with a variable of type `int`.

This violates the stub authoring guideline: parameter types must not be stricter than Laravel's actual signatures.

## Reproduction

Found while auditing [monicahq/monica](https://github.com/monicahq/monica) at Psalm errorLevel 1.

**File:** `app/Console/Commands/Local/MonicaLocalize.php:174-176`

```php
$strings = collect(['zero', 'one', 'few', 'many', 'other']);

if ($j >= $strings->count()) {
    // $strings->count() - 1 inferred as int<-1, max>
    $message = $strings[$strings->count() - 1]; // ArgumentTypeCoercion
} elseif (! $result->has($j)) {
    // $j is int
    $message = $strings[$j]; // ArgumentTypeCoercion
}
```

Psalm errors:
- `Argument 1 of Collection::offsetGet expects int<0, max>, but int<-1, max> provided`
- `Argument 1 of Collection::offsetGet expects int<0, max>, but parent type int provided`

## Expected Behavior

No error. Laravel's `Collection::offsetGet` accepts any key type — it doesn't validate that the offset is non-negative.

## Proposed Fix

Change the stub parameter from `int<0, max>` to `int` (or the appropriate key type `TKey`). The narrower `int<0, max>` would only be acceptable as a *return* type, not a parameter type.
